pymatgen/command_line/bader_caller.py

```diff
@@ -193,3 +206,175 @@ def get_oxidation_state_decorated_structure(self):
                    for i in range(len(structure))]
         structure.add_oxidation_state_by_site(charges)
         return structure
+
+    @property
+    def result(self):
+
+        summary = {
+            "min_dist": [d['min_dist'] for d in self.data],
+            "charge": [d['charge'] for d in self.data],
+            "volume": [d['atomic_vol'] for d in self.data],
+            "vacuum_charge": self.vacuum_charge,
+            "vacuum_volume": self.vacuum_volume,
+            "reference_used": self.reference_used,
+            "bader_version": self.version,
+        }
+
+        if self.potcar:
+            charge_transfer = [self.get_charge_transfer(i) for i in range(len(self.data))]
+            summary['charge_transfer'] = charge_transfer
+
+        return BaderResult(summary)
+
+
+class BaderResult(MSONable):
```

Comment:
Point well taken about MSONable -- will leave them as plain dicts for now, and possibly refactor in future if it calls for it. Have moved the `from_path` and `from_object` methods into their own standalone functions.